alchemy/src/aws/utils.ts

```diff
@@ -0,0 +1,137 @@
+import { fromNodeProviderChain } from "@aws-sdk/credential-providers";
+import { loadConfig } from "@smithy/node-config-provider";
+import { AwsClient } from "aws4fetch";
```

Comment:
Interesting choice to use aws4fetch instead of the aws-sdk.

I am actually exploring a similar approach with [itty-aws](https://github.com/sam-goodwin/itty-aws) except also with effect.

Reply:
I chose this approach largely because you've expressed a preference for http-based invocation than bundling clients before. 

I do think itty-aws is probably a nice way to go, and I forgot about it when I started this implementation.